### DISCUSSION TITLE
Instancia de um Objeto

### DISCUSSION THREAD

**Discussion by Themystocles** [balta-io/forum#268] [Tira Dúvidas] [answered]
Eu Crio uma classe chamada " Caneta "  .  No meu metodo main eu instancio essa classe -> var caneta = new caneta();   e faço umas atribuições nas propriedades que eu havia definido:
caneta.nome = "bick";
caneta.ponta = 5.0;

Nesse contexto criado, caso eu crie outra classe qualquer e queira puxar dentro dessa classo o "Caneta.nome" por exemplo.  É possível fazer isso. ?  Eu fiz algumas tentativas sem sucesso. pra eu usar o objeto caneta dentro de outra classe eu tive que instanciar novamente na classe que eu queria que fosse usada e definir novamente os atributos.

minha dúvida é se é possivel usar o objeto criado no main dentro de qualquer outra classe.

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @Themystocles, como vai?
Quais tentativas você fez que não obteve sucesso? Pergunto pois assim posso te orientar melhor aqui.

**Comment by Themystocles** [comment]
![Exemplo](https://user-images.githubusercontent.com/41810509/219634306-d3b6bc8e-f3cc-433e-86cf-9b67a32cc0e4.png)


Veja!  Minha dificuldade é chamar esse objeto dentro de uma classe. Estou usando o dotnet 7. Eu gostaria de saber se estou chamando corretamente.   Meu atributo "nome" está como publico, eu consigo chamar ele no main após instanciar, mas dentro de qualquer outra classe eu não consigo chamar sem fazer outra instancia.

**Reply by andrebaltieri** [reply]
Entendi @Themystocles, isto não é possível desta forma, nem só por conta do objeto não ser estático, mas também pela classe não permitir execução direta (Sem ser via um método).

Qual curso/aula está fazendo que surgiu esta dúvida?

**Reply by Themystocles** [reply]
Hm... Entendi.   a duvida surgiu na Aula de "Propiedades"   Após a aula eu comecei a brincar com as propriedades de uma classe que eu criei  foi então que eu percebi que não conseguia chamar em outra classe.

**Reply by andrebaltieri** [reply]
Entendi... segue o curso de C#, depois de OOP, que vou explicar tudo em detalhes neles 🚀
Parece complexo no começo, mas depois você pega o jeito...

**Reply by Themystocles** [reply]
Lega.  Muito obrigado, vou seguir com o curso!!
